src/bin/casr-cli.rs

```diff
@@ -629,3 +660,215 @@ fn change_text_view(layout1: &mut LinearLayout, act: Action) -> Option<EventResu
         Action::Mouse(_) => None,
     }
 }
+
+/// Function prints summary after casr work
+///
+/// # Arguments
+///
+/// * 'casr_dir' - directory with clusters
+///
+fn print_summary(casr_dir: PathBuf) -> Result<()> {
+    // Hash each class in whole casr directory
+    let mut casr_classes: HashMap<String, i32> = HashMap::new();
+
+    let mut clusters: Vec<(PathBuf, i32)> = Vec::new();
+    for cl_path in fs::read_dir(&casr_dir).unwrap().flatten() {
+        let cluster = cl_path.path();
+        let filename = cluster.file_name().unwrap().to_str().unwrap();
+
+        // check dir name
+        if !filename.starts_with("cl") || !cluster.is_dir() || filename.starts_with("clerr") {
+            continue;
+        } else {
+            clusters.push((cluster.to_path_buf(), filename[2..].parse::<i32>().unwrap()));
+        }
+    }
+    clusters.sort_by(|a, b| a.1.cmp(&b.1));
+    if clusters.is_empty()
+        && fs::read_dir(&casr_dir)
+            .unwrap()
+            .filter(|res| res.is_ok())
+            .map(|res| res.unwrap().path())
+            .any(|e| e.extension().is_some() && e.extension().unwrap() == "casrep")
+    {
+        clusters.push((casr_dir, 0));
+    }
+
+    for cl_path in clusters {
+        let cluster = cl_path.0.as_path();
+        let filename = cluster.file_name().unwrap().to_str().unwrap();
+
+        info!("==> <{}>", filename.magenta());
+
+        // Hash each crash in cluster
+        let mut cluster_hash: HashMap<String, (Vec<String>, i32)> = HashMap::new();
+        // Hash each class in cluster
+        let mut cluster_classes: HashMap<String, i32> = HashMap::new();
+        // Hash files
+        let mut filestems: HashSet<PathBuf> = HashSet::new();
+        for report_path in fs::read_dir(cluster).unwrap().flatten() {
+            let report = report_path.path();
+            if report.is_file()
+                && report.extension().is_some()
+                && report.extension().unwrap() == "casrep"
+            {
+                // report == .*/crash-hash.gdb.casrep
```

Comment:
maybe just crash.gdb.casrep? It is not always libFuzzer format. It is just an example